### PR TITLE
RM#71613: TRANSLATIONS : Export/import translations

### PR DESCRIPTION
**Pull request by tyu-axelor**

### CODE REVIEW

**Review by jmi-axelor — APPROVED** [review]

File: axelor-base/src/main/java/com/axelor/apps/base/web/ImportExportTranslationController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    Path path = null;
+    try {
+      path = importExportTranslationService.importTranslations(importExportTranslation);
+    } catch (AxelorException e) {
+      Logger logger = LoggerFactory.getLogger(getClass());
+      logger.error("Read CSV file failed.", e);
+      TraceBackService.trace(response, e, ResponseMessageType.ERROR);
+    }
+    if (path == null) {
+      response.setInfo("The import file is empty or it has error format.");
```
Comment (jmi-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/web/ImportExportTranslationController.java:57]: This message can't be translated, please check other controllers to see how to make it translatable

File: axelor-base/src/main/java/com/axelor/apps/base/web/ImportExportTranslationController.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      response.setReload(true);
+    } else {
+      response.setInfo("File successfully imported.");
```
Comment (jmi-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/web/ImportExportTranslationController.java:60]: This message can't be translated, please check other controllers to see how to make it translatable

File: axelor-base/src/main/java/com/axelor/apps/base/service/ImportExportTranslationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+        JPA.em()
+            .createQuery(
+                "select mt.language, mt.key, mt.message "
+                    + "from MetaTranslation mt "
+                    + "where mt.language = ANY ( "
+                    + "Select ls.code "
+                    + "from ImportExportTranslation iet "
+                    + "left join iet.languageSet ls "
+                    + "where iet.id = :curId"
+                    + ") ")
```
Comment (jmi-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/service/ImportExportTranslationServiceImpl.java:258]: add an order by mt.id just to be sure

**Review by jmi-axelor — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/71613.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+title: "TRANSLATIONS : Export/import translations"
+type: feature
+description: |
+  Create a process to export and import translations in different languages.
+  A new field called isNative is added to each language obj.
+  The native ones are English and French, which cannot be exported or imported.
+  Other available languages can be exported into a csv file, and can be modified by user after downloading.
```
Comment (jmi-axelor) [inline changelogs/unreleased/71613.yml:8]: Change these lines, native languages don't exist anymore

File: axelor-base/src/main/resources/apps/demo-data/fr/base_language.csv [hunk 5]
```diff
@@ -1,3 +1,3 @@
 "importId";"name";"code"
 1;"Anglais";"en"
-2;"Français";"fr"
+2;"Français";"fr"
```
Comment (jmi-axelor) [inline axelor-base/src/main/resources/apps/demo-data/fr/base_language.csv:3]: try to remove this modification, nothing should change in this file

File: axelor-base/src/main/resources/data-init/input/base_language.csv [hunk 6]
```diff
@@ -1,3 +1,3 @@
 "importId";"name";"code"
 1;"English";"en"
-2;"French";"fr"
+2;"French";"fr"
```
Comment (jmi-axelor) [inline axelor-base/src/main/resources/data-init/input/base_language.csv:3]: try to remove this modification, nothing should change in this file

**Review by jmi-axelor — CHANGES_REQUESTED** [review]

**Review by ale-axelor — CHANGES_REQUESTED** [review]

**Pull request merged**